### PR TITLE
Auto-initialize DVC repo

### LINKED ISSUE #522: dvc: Catch `Empty Git repo` error.
This can occur when using `save_dvc_exp=True` on an empty repo:

```console
$ git init
$ dvc init
$ python -c "from dvclive import Live;Live(save_dvc_exp=True)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/daviddelaiglesiacastro/miniconda3/lib/python3.9/site-packages/dvclive/live.py", line 79, in __init__
    self._init_dvc()
  File "/Users/daviddelaiglesiacastro/miniconda3/lib/python3.9/site-packages/dvclive/live.py", line 132, in _init_dvc
    self._baseline_rev = self._dvc_repo.scm.get_rev()
  File "/Users/daviddelaiglesiacastro/miniconda3/lib/python3.9/site-packages/scmrepo/git/__init__.py", line 286, in _backend_func
    result = func(*args, **kwargs)
  File "/Users/daviddelaiglesiacastro/miniconda3/lib/python3.9/site-packages/scmrepo/git/backend/dulwich/__init__.py", line 395, in get_rev
    raise SCMError("Empty git repo")
scmrepo.exceptions.SCMError: Empty git repo
```

### PR DESCRIPTION
**Pull request by dberenbaum**
Related to #531. Do you think it's enough @daavoo?

Example with no Git repo (with or without a DVC repo):

```
WARNING:dvclive:Can't save experiment without a Git Repo.
Create a Git repo (`git init`) and commit (`git commit`).
```

Example with an empty Git repo and no DVC repo:

```
Initialized DVC repository.

INFO:dvc.repo.init:Initialized DVC repository.

You can now commit the changes to git.

INFO:dvc.repo.init:You can now commit the changes to git.

WARNING:dvclive:Can't save experiment to an empty Git Repo.
Add a commit (`git commit`) to save experiments.
INFO:dvclive:To run with DVC, add this to /private/tmp/repo/dvc.yaml:
stages:
  dvclive:
    cmd: <python my_code_file.py my_args>
    deps:
    - <my_code_file.py>
```

Example with an empty Git repo and an existing DVC repo:

```
WARNING:dvclive:Can't save experiment to an empty Git Repo.
Add a commit (`git commit`) to save experiments.
INFO:dvclive:To run with DVC, add this to /private/tmp/repo/dvc.yaml:
stages:
  dvclive:
    cmd: <python my_code_file.py my_args>
    deps:
    - <my_code_file.py>
```

Example with a non-empty Git repo and no DVC repo:

```
Initialized DVC repository.

INFO:dvc.repo.init:Initialized DVC repository.

You can now commit the changes to git.

INFO:dvc.repo.init:You can now commit the changes to git.

INFO:dvclive:To run with DVC, add this to /private/tmp/repo/dvc.yaml:
stages:
  dvclive:
    cmd: <python my_code_file.py my_args>
    deps:
    - <my_code_file.py>
```

Edit: also fixes #522

### CODE REVIEW

File: src/dvclive/dvc.py [hunk 1]
```diff
@@ -77,12 +77,16 @@ def make_checkpoint():
 def get_dvc_repo():
     from dvc.exceptions import NotDvcRepoError
     from dvc.repo import Repo
-    from dvc.scm import SCMError
+    from dvc.scm import Git, SCMError
+    from scmrepo.exceptions import SCMError as GitSCMError
```
Comment (daavoo) [inline src/dvclive/dvc.py:81]: I am a little confused by the 2 SCMError, I am guessing one inherits from DvcException and the other not?

**Review by daavoo — COMMENTED** [review]
The code changes look good to me but the duplicated logging by the DVC `Repo` API is starting to bother me.

I forgot to open an issue for the duplicated `experiments.save` warnings but this P.R. showcases the problem even more.

@dberenbaum if its okay, I would prefer to make a release with the other changes we recently merged and wait for this one, as I didn't have time to really test it

**Comment by dberenbaum** [comment]
That's fine with me @daavoo

**Review by daavoo — APPROVED** [review]

File: src/dvclive/dvc.py (hunk 1 above)
Comment (dberenbaum) [reply src/dvclive/dvc.py:81]: Yes. This felt not too clean to me but didn't immediately know a better way to handle it.

**Comment by dberenbaum** [comment]
@daavoo Sorry if I wasn't clear. I meant I am fine with trying to rely on DVC here instead of duplicating logic in DVCLive.

**Comment by dberenbaum** [comment]
Got it. I think we just need to drop `logging.basicConfig()` because it's calling the root logger.

**Pull request merged**